### PR TITLE
Refs [TASK][34635] Handle event Hourly Forecast Screen

### PR DESCRIPTION
**Pull request by I-am-Lien**
## Related Tickets
- [#Task 34635: Content](https://edu-redmine.sun-asterisk.vn/issues/34635)
## WHAT
- Handle event Hourly Forecast Screen
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/li/weatherapp/data/model/Hourly.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        val temp: Double,
+        val feelsLike: Double,
+        val pressure: Long,
+        val humidity: Long,
+        val windSpeed: Double,
+        val windDeg: Long,
+        val weather: Weather,
+        val clouds: Long,
+        val uvi: Double
+) : Parcelable {
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/data/model/Hourly.kt:19]: space là 4 thôi

Comment (huytq-2422) [reply app/src/main/java/com/li/weatherapp/data/model/Hourly.kt:19]: indent là 4 sapce thôi ông

File: app/src/main/java/com/li/weatherapp/ui/hourly/HourlyForecastFragment.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import kotlinx.android.synthetic.main.fragment_hourly_forecast.*
+
+class HourlyForecastFragment : BaseFragment(), HourlyForeCastContract.View {
+
+    private var presenter: HourlyForeCastContract.Presenter? = null
+    private val adapter = HourlyForecastAdapter(this::hourlyForecastItemClick)
+
+    override val layoutResource get() = R.layout.fragment_hourly_forecast
+
+
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/hourly/HourlyForecastFragment.kt]: không để 2 dòng trống liên tiếp

File: app/src/main/java/com/li/weatherapp/utils/LongExtention.kt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package com.li.weatherapp.utils
+
+fun Long.convertToMilli(): Long = this * Constants.MILLI_TO_SECOND
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/utils/LongExtention.kt]:
```suggestion
fun Long.convertToMilli(): Long = this *  DateUtils.SECOND_IN_MILLIS
```

File: app/src/main/java/com/li/weatherapp/ui/hourly/HourlyForecastFragment.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        recyclerViewHourly.adapter = adapter
+    }
+
+    private fun hourlyForecastItemClick(item: Hourly) {
+        val hourlyBottomSheetFragment = HourlyBottomSheetFragment.newInstance(item)
+        hourlyBottomSheetFragment.show(childFragmentManager, TAG)
+    }
+
+    companion object {
+        const val TAG = "HourlyBottomSheet"
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/hourly/HourlyForecastFragment.kt:75]: Tại sao phải dùng TAG thế ?

File: app/src/main/java/com/li/weatherapp/ui/adapters/HourlyViewHolder.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        private val view: View,
+        private var itemClick: (Hourly) -> Unit
+) : BaseViewHolder<Hourly>(view) {
+    private var itemHourly: Hourly? = null
+
+    init {
+        view.setOnClickListener {
+            itemHourly?.let {
+                itemClick(it)
+            }
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/adapters/HourlyViewHolder.kt]:
```suggestion
            itemHourly?.let (itemClick(it))
```

File: app/src/main/java/com/li/weatherapp/ui/hourly/HourlyForecastFragment.kt (hunk 4 above)
Comment (I-am-Lien) [reply app/src/main/java/com/li/weatherapp/ui/hourly/HourlyForecastFragment.kt:75]: Nó là tham số truyền vào function để phân biệt các bottomSheet với nhau ô

**Pull request merged**